pkg/webhooks/webhookpatch.go

```diff
@@ -50,64 +52,79 @@ type WebhookCertPatcher struct {
 	revision    string
 	webhookName string
 
-	queue queue.Instance
+	queue workqueue.RateLimitingInterface
 
 	// File path to the x509 certificate bundle used by the webhook server
 	// and patched into the webhook config.
 	CABundleWatcher *keycertbundle.Watcher
-	whcLw           *cache.ListWatch
-}
 
-// Run runs the WebhookCertPatcher
-func (w *WebhookCertPatcher) Run(stopChan <-chan struct{}) {
-	go w.queue.Run(stopChan)
-	go w.runWebhookController(stopChan)
-	go w.startCaBundleWatcher(stopChan)
+	informer cache.SharedIndexInformer
 }
 
 // NewWebhookCertPatcher creates a WebhookCertPatcher
 func NewWebhookCertPatcher(
-	client kubernetes.Interface,
+	client kubelib.Client,
 	revision, webhookName string, caBundleWatcher *keycertbundle.Watcher) (*WebhookCertPatcher, error) {
-	whcLw := cache.NewFilteredListWatchFromClient(
-		client.AdmissionregistrationV1().RESTClient(),
-		"mutatingwebhookconfigurations",
-		"",
-		func(options *metav1.ListOptions) {
-			options.LabelSelector = fmt.Sprintf("%s=%s", label.IoIstioRev.Name, revision)
-		})
-	retryBackoff := backoff.NewExponentialBackOff()
-	retryBackoff.InitialInterval = 2 * time.Second
-	retryBackoff.MaxInterval = 15 * time.Minute // Max Backoff interval.
-	retryBackoff.MaxElapsedTime = 0             // Never stop retrying.
-	return &WebhookCertPatcher{
+	p := &WebhookCertPatcher{
 		client:          client,
 		revision:        revision,
 		webhookName:     webhookName,
 		CABundleWatcher: caBundleWatcher,
-		queue:           queue.NewBackOffQueue(retryBackoff),
-		whcLw:           whcLw,
-	}, nil
+		queue:           workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "mutatingwebhookconfiguration"),
+	}
+	informer := admissioninformer.NewFilteredMutatingWebhookConfigurationInformer(client, 0, cache.Indexers{}, func(options *metav1.ListOptions) {
+		options.LabelSelector = fmt.Sprintf("%s=%s", label.IoIstioRev.Name, revision)
+	})
+	p.informer = informer
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			oldConfig := oldObj.(*v1.MutatingWebhookConfiguration)
+			newConfig := newObj.(*v1.MutatingWebhookConfiguration)
+			p.updateWebhookHandler(oldConfig, newConfig)
```

Comment:
nit: I think its slightly more idiomatic k8s controller to just get the key (name) of newObj here and enqueue it. The queue will handle the things in updateWebhookHandler (skipping if no update is needed).

I don't see any issues with this - should work - just nitpicks